src/ome_zarr_models/_rfc5_transforms/multiscales.py

```diff
@@ -0,0 +1,247 @@
+    ) -> object:
+        """
+        Ensures that
+        - a single transformation is present
+        - such transformation is a scale
+        - if such transformation is a sequence, ensure that its length is 2 and that
+          the first transformation is a scale and the second a translation
+        """
+
+        class Transforms(BaseModel):
+            transforms: tuple[CoordinateTransformationType, ...]
+
+        transforms = Transforms(transforms=transforms_obj).transforms
+        check_length(transforms, valid_lengths=[1], variable_name="transforms")
+
+        transform = transforms[0]
+        if transform.type == "sequence":
+            check_length(
+                transform.transformations,
+                valid_lengths=[2],
+                variable_name="transform.transforms (i.e. transformations composing "
+                "the sequence)",
+            )
+            first, second = transform.transformations
+            if first.type != "scale":
+                msg = (
+                    "When the first (and only) element in `coordinateTransformations`"
+                    " is a `Sequence`, the first element must be a `Scale` transform. "
+                    f"Got {first} instead."
+                )
+                raise ValueError(msg)
+            if second.type != "translation":
+                msg = (
+                    "When the first (and only) element in `coordinateTransformations`"
+                    " is a `Sequence`, the second element must be a `Translation` "
+                    f"transform. Got {second} instead."
+                )
+                raise ValueError(msg)
+        elif transform.type != "scale":
+            msg = (
+                "The first transformation in `coordinateTransformations` "
+                "must either be a `Scale` transform or a `Sequence` transform. "
+                f"Got {transform} instead."
+            )
+            raise ValueError(msg)
+
+        return transforms_obj
+
+    @field_validator("coordinateTransformations", mode="after")
+    @classmethod
+    def _ensure_transform_dimensionality(
+        cls,
+        transforms: tuple[CoordinateTransformationType, ...],
+    ) -> tuple[CoordinateTransformationType, ...]:
+        """
+        Ensures that the dimensionality of the scale and translation (when both present)
+        match
+        """
+        maybe_sequence = transforms[0]
+        if maybe_sequence.type == "sequence":
+            first, second = maybe_sequence.transformations
+            if (
+                isinstance(first, Scale)
+                and isinstance(second, Translation)
+                and len(first.scale) != len(second.translation)
+            ):
+                raise ValueError(
+                    "The length of the scale and translation vectors must be the same."
+                    f"Got {len(first.scale)} and {len(second.translation)}."
+                )
+        return transforms
```

Comment:
is this dimensionality validation not already part of the sequence class definition? i.e., once we know it's a sequence, haven't we already checked that the transforms inside the sequence are dimensionally compatible?

Reply:
Good point. I implemented this because `Sequence` is still empty, but it's better to move the validation there.